requirements-dev.txt

```diff
@@ -2,9 +2,8 @@ git+https://github.com/astropy/photutils.git
 git+https://github.com/spacetelescope/stsci.tools.git
 git+https://github.com/spacetelescope/stwcs.git
 git+https://github.com/spacetelescope/stregion.git
-#git+https://github.com/astropy/astroquery.git#egg=astroquery
---extra-index-url https://pypi.anaconda.org/astropy/simple astropy --pre
-
---extra-index-url https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
```

Comment:
[compat-openssl11](https://github.com/spacetelescope/drizzlepac/pull/1714/files/967c04c363efba66bb7aabb37c42b74cb067eec3#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449R52)

I think I managed to avoid putting these in the file by adjusting the index for the `devdeps` target. The extra arguments were causing tox's parser to fail. I'll double-check, though I think adding the extra index then asking for `>=0.0.dev0` is roughly equivalent to setting  pip install -Ihttps://index/here --pre`.

Reply:
Oh I see. Were these our packages or third party?

Reply:
I wish I'd kept the logs. I'll find out.
Meanwhile you helped me realized I'd forgotten the `.dev0` suffix on the `numpy` line... So that's fixed.

Reply:
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. contourpy 1.2.0 requires numpy<2.0,>=1.20, but you have numpy 2.0.0.dev0 which is incompatible.
matplotlib 3.8.2 requires numpy<2,>=1.21, but you have numpy 2.0.0.dev0 which is incompatible.
pandas 2.1.4 requires numpy<2,>=1.22.4; python_version < "3.11", but you have numpy 2.0.0.dev0 which is incompatible.
scikit-learn 1.3.2 requires numpy<2.0,>=1.17.3, but you have numpy 2.0.0.dev0 which is incompatible.
```

If I satisfy the requirements by installing the development versions for the packages above, pytest greets me with an endless wall of text telling me to rebuild all packages from source that haven't been linked to numpy 2.0, followed by a stream of ABI errors:

```python
RuntimeError: module compiled against ABI version 0x1000009 but this version of numpy is 0x2000000
```

Drizzlepac has so many dependencies. Pip's incompatibility warnings only display packages with explicit pins (`<2.0`). It doesn't account for wheels providing packages with incompatible linkage.